OData/src/System.Web.OData/OData/Builder/PrimitivePropertyConfigurationExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Web.Http;
+using System.Web.OData.Properties;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Library;
+
+namespace System.Web.OData.Builder
+{
+    /// <summary>
+    /// Extensions method for <see cref="PrimitivePropertyConfiguration"/>.
+    /// </summary>
+    public static class PrimitivePropertyConfigurationExtensions
+    {
+        /// <summary>
+        /// If this primitive property is <see cref="System.DateTime"/>, this method will make the target
+        /// Edm type kind as <see cref="Date"/>
+        /// </summary>
+        /// <param name="property">Reference to the calling primitive property configuration.</param>
+        /// <returns>Returns itself so that multiple calls can be chained.</returns>
+        public static PrimitivePropertyConfiguration AsDate(this PrimitivePropertyConfiguration property)
```

Comment:
The api does not look correct.  A primitive property configuration can be anything.  We should enable user to write correctly easier, not another way around - with run time exceptions.